aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/util/ImmutabilityHelper.java

```diff
@@ -42,12 +43,18 @@ static boolean isEngineMutable(final ResourceModel previous, final ResourceModel
 
     static boolean isPerformanceInsightsKMSKeyIdMutable(final ResourceModel previous, final ResourceModel desired) {
 
-        return StringUtil.isNullOrEmpty(previous.getPerformanceInsightsKMSKeyId()) ||
+        return StringUtils.isNullOrEmpty(previous.getPerformanceInsightsKMSKeyId()) ||
                 Objects.equal(previous.getPerformanceInsightsKMSKeyId(), desired.getPerformanceInsightsKMSKeyId());
     }
 
+    static boolean isAvailabilityZoneChangeMutable(final ResourceModel previous, final ResourceModel desired) {
+        return Objects.equal(previous.getAvailabilityZone(), desired.getAvailabilityZone()) ||
```

Comment:
Should this not be: `!Objects.equal(previous.getAvailabilityZone(), desired.getAvailabilityZone())`

Reply:
No, we cannot change SingleAZ instance's availability zone without replacement